doc/nrf/libraries/bootloader/bl_monotonic_counters.rst

```diff
@@ -0,0 +1,47 @@
+.. _doc_bl_monotonic_counters:
+
+Monotonic counters
+##################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The monotonic counters library implements the logic for readind and writing uint16_t counters which can only increase.
+
+The library is used by :ref:`bootloader` to keep track of the version of the application.
+The monotonic counters are stored as part of the bootloader storage :ref:`doc_bl_storage`.
+When updating the application, the :ref:`doc_bl_validation` library checks if the new version is newer than the current version and only accepts the new image if it is.
+After updating, the application must then store the new version information in the bootloader storage.
```

Comment:
```suggestion
After updating, the application must then store the new version information in the bootloader storage.
```

Which application? MCUBoot, or the image that is booted by MCUBoot?

Reply:
This is surprising, I assumed it would be NSIB that updated the version after validating the image it's going to boot.

Is this delayed counter update to avoid images that don't boot?

Reply:
This is part of the old documentation. Only NSIB and MCUBOOT can update the version.